### PR TITLE
implemented get/api to retrieve all endpoints

### PR DESCRIPTION
**Pull request by merongb**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -26,4 +28,19 @@ describe('GET /api/topics', () => {
...
+});
+
+describe('GET /api', () => {
+    test('returns a 200 status code', () => {
+        return request(app).get("/api").expect(200)
+    });
+    test('returns a json object matching of all the endpoints.json file', () => {
+        return request(app).get("/api").expect(200).then(({body}) => {
+            expect(body).toEqual(endpoints)
+            expect(body).to
```
Comment (Augs0) [inline __tests__/app.test.js:40]: Looks like an unfinished test here?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -26,4 +28,19 @@ describe('GET /api/topics', () => {
...
+        })
+    });
+    test('returns a 404 status code when given a bad api request', () => {
```
Comment (Augs0) [inline __tests__/app.test.js:43]: We don't need a further 404 test. It should cover all endpoints that do not exist and you have already tested for this previously

File: db/app.controllers.js [hunk 3]
```diff
@@ -7,4 +9,9 @@ exports.getTopics = (req, res, next) => {
     .catch((err) => {
         next(err)
     })
-}
\ No newline at end of file
+}
+
+exports.getEndpoints = (req, res) => {
+   res.send(endpoints)
```
Comment (Augs0) [inline db/app.controllers.js:15]: I think you mentioned this in your submission but don't forget to add a status code

File: db/app.js [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
+
 
 app.use((err, req, res, next) => {
     console.log(err)
     res.status(500).send({message : "Server error"})
 })
 
 app.all("/*", (req, res, next) => {
-    res.status(404).send({ message : "Path Not Found!"})
+    res.status(404).send("Path Not Found!")
```
Comment (Augs0) [inline db/app.js:16]: It's fine to use message here, you just need to make sure you access it properly in the test

**Review by Augs0 — APPROVED** [review]
Mostly a few things to remove, Meron, but overall you have hit the objectives. Take your time and be sure you have completed everything before submitting to avoid small mistakes. You can merge once you've added the status code in your controller.

File: __tests__/app.test.js (hunk 1 above)
Comment (merongb) [reply __tests__/app.test.js:40]: was meant to delete that

File: __tests__/app.test.js (hunk 2 above)
Comment (merongb) [reply __tests__/app.test.js:43]: noted

**Pull request merged**